src/test/java/integration/toyprojects/R3ToyProjectTest.java

```diff
@@ -95,17 +95,17 @@ public void stableThresholds() {
 		//Manually Verified
 		List<StableCommit> stableCommitLow = getStableCommits().stream().filter(commit ->
 				commit.getCommitThreshold() == 3).collect(Collectors.toList());
-		Assert.assertEquals(3, stableCommitLow.size());
+		Assert.assertEquals(9, stableCommitLow.size());
```

Comment:
the numbers should still be the same, or no?

Reply:
We are now collecting more stable instances because we collect a stable instance once for each applicable level, e.g. if an instance has methods with variables, we now collect instances for the class level, the method level, and the variable level. This creates lots of duplicate entries in the DB, but simplifies the usage and understanding a lot.